opm/input/eclipse/Schedule/Group/GuideRate.cpp

```diff
@@ -132,6 +133,15 @@ double Opm::GuideRate::get(const std::string& name, const Phase& phase) const
     return iter->second;
 }
 
+const Opm::GuideRate::RateVector& Opm::GuideRate::getPotentials(const std::string& name) const
+{
+    if (!this->hasPotentials(name)) {
+        auto message = fmt::format("Potentials for '{}' do not exist.", name);
+        throw std::logic_error {message};
+    }
+    return this->potentials.at(name);
+}
```

Comment:
You are triggering two searches here: one in hasPotentials and the other one ins `at`. In addition there is double error handling as `at` does another check.

If you want the better error message then we could do:
```suggestion
    if (const auto candidate = this->potentials.find(name); candidate == this->potentials.end()) {
        auto message = fmt::format("Potentials for '{}' do not exist.", name);
        throw std::logic_error {message};
    } else {
        return candidate->second;
    }
}
```

Reply:
Good idea. I have updated the PR accordingly.